tooling/noir_js_backend_barretenberg/src/public_inputs.ts

```diff
@@ -1,7 +1,7 @@
 import { Abi, WitnessMap } from '@noir-lang/types';
 
 export function flattenPublicInputs(publicInputs: WitnessMap): string[] {
-  const publicInputIndices = [...publicInputs.keys()].sort();
+  const publicInputIndices = [...publicInputs.keys()].sort((a, b) => a - b);
```

Comment:
For a test, we would just need to have a circuit with more than ten input inputs I think

Reply:
I've added a test which covers most of the potential pitfalls:
- inputs with indices 2 and 13 to hit ascii ordering.
- duplicated public inputs in the return values.
- an input being returned as a return value also.

Reply:
perfecto ser